server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -291,6 +301,21 @@ private CompletionStage<RestResponse> memory(RestRequest restRequest) {
       return asJsonResponseFuture(new JVMMemoryInfoInfo().toJson());
    }
 
+   private CompletionStage<RestResponse> heapDump(RestRequest restRequest) {
+      boolean live = Boolean.parseBoolean(restRequest.getParameter("live"));
+      ServerManagement server = invocationHelper.getServer();
+      return CompletableFuture.supplyAsync(() -> {
+         try {
+            Path dumpFile = Files.createTempFile(server.getServerDataPath(), "dump", ".hprof");
+            Files.delete(dumpFile);
+            new JVMMemoryInfoInfo().heapDump(dumpFile, live);
+            return asJsonResponse(Json.object().set("path", dumpFile.getFileName().toString()));
```

Comment:
Maybe just name if `fileName` now?

Reply:
To clarify should the JSON name be `path` or `fileName`? Otherwise I am ready to integrated.

Reply:
I've made it `filename`